### DISCUSSION TITLE
Backward compatible of compound datasets

### DISCUSSION THREAD

**Discussion by Julian9620** [LiorBanai/HDF5-CSharp#182] [Show and tell]
I tested it and i know now: If there was an update of the struct which represents a row you can add a new column for the new data. If you load now the old data does not matter.
Tests were with: 
`
    [Serializable]
    [StructLayout(LayoutKind.Sequential)]
    public struct Row
    {
        [Hdf5EntryName("doublevalue")]
        public double doublevalue;
        [Hdf5EntryName("stringvalue")]
        [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 64)]
        public string stringvalue;
    }
`
to:
`
    [Serializable]
    [StructLayout(LayoutKind.Sequential)]
    public struct Row
    {
        [Hdf5EntryName("doublevalue")]
        public double doublevalue;
        [Hdf5EntryName("stringvalue")]
        [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 64)]
        public string stringvalue;
        [Hdf5EntryName("intvalue")]
        [MarshalAs(UnmanagedType.U4)]
        public uint intvalue;
    }
`
Great Job @LiorBanai 👍

**Comment by LiorBanai** [comment]
I'm glad the library meets your needs 🙂
